scripts/contractInteraction/loan_tokens.py

```diff
@@ -339,4 +345,59 @@ def readLiquidity():
     
     tokenContract = Contract.from_abi("Token", address=conf.contracts['WRBTC'], abi=TestToken.abi, owner=conf.acct)
     bal = tokenContract.balanceOf(conf.contracts['ConverterUSDT'])
-    print("supply of rBTC on swap", bal/1e18)
\ No newline at end of file
+    print("supply of rBTC on swap", bal/1e18)
+
+
+# Notes: This function will do:
+# 1. tradeOpenAndClosingWithoutCollateral (using amountUnderlying that is sent)
+# 2. lendToPool (using amountUnderlying that is sent)
+# 3. removeFromPool (50% of the lending)
+# 4. tradeOpenAndClosingWithCollateral (using amountCollateral that is sent)
+# 5. Test borrow (using amountCollateral that is sent)
+#
+# WARN:
+# 1. make sure you have 2 times balance of underlyingTokenAddress
+# 2. make sure you have 2 times balance of amountCollateral
+def wrappedIntegrationTest(loanTokenAddress, underlyingTokenAddress, collateralTokenAddress, amountUnderlying, amountCollateral):
+    sovryn = Contract.from_abi("sovryn", address=conf.contracts['sovrynProtocol'], abi=interface.ISovrynBrownie.abi, owner=conf.acct)
+    loanToken = Contract.from_abi("loanToken", address=loanTokenAddress, abi=LoanTokenLogicStandard.abi, owner=conf.acct)
+
+    underlyingToken = Contract.from_abi("TestToken", address=underlyingTokenAddress, abi=ERC20.abi, owner=conf.acct)
+    collateralToken = Contract.from_abi("TestToken", address=collateralTokenAddress, abi=ERC20.abi, owner=conf.acct)
+    
+    prevUnderlyingBalance = underlyingToken.balanceOf(conf.acct)
+    prevCollateralBalance = collateralToken.balanceOf(conf.acct)
+
+    # ------------------------------------------------ Test Trade Open & Close without collateral ------------------------------------------------------
+    print("Test Trade open and closing without collateral")
+    testTradeOpeningAndClosing(sovryn.address, loanToken.address, underlyingTokenAddress, collateralTokenAddress, amountUnderlying, 2e18, True,0)
+
+    print("=============================== PREVIOUS BALANCE ====================================")
+    print("Underlying balance: ", prevUnderlyingBalance)
+    print("Collateral balance: ", prevCollateralBalance)
+
+    print("=============================== UPDATED BALANCE =====================================")
+    updatedUnderlyingBalance = underlyingToken.balanceOf(conf.acct)
+    updatedCollateralBalance = collateralToken.balanceOf(conf.acct)
+    print("Underlying balance: ", updatedUnderlyingBalance)
+    print("Collateral balance: ", updatedCollateralBalance)
+
+    # Verify
+    if prevUnderlyingBalance - updatedUnderlyingBalance != amountUnderlying:
+        raise Exception("Updated underyling balance is not matched with the amount that was traded")
```

Comment:
`print('{0} {1} cost ${2}'.format(6, 'bananas', 1.74)`
`raise Exception('Updated underyling balance: {0} should match the amount that was traded {1}'.fomat(prevUnderlyingBalance - updatedUnderlyingBalance, amountUnderlying))`